velox/functions/prestosql/tests/TimestampWithTimeZoneCastTest.cpp

```diff
@@ -64,6 +65,74 @@ TEST_F(TimestampWithTimeZoneCastTest, fromTimestamp) {
   testCast(tsVector, expected);
 }
 
+TEST_F(TimestampWithTimeZoneCastTest, fromVarchar) {
+  const auto stringVector = makeNullableFlatVector<StringView>(
```

Comment:
I just merged a PR that lets you use evaluateOnce() with custom types. Could you rebase and use that instead so you could specify one test case per line of code?

Reply:
Would that work here? As I'm testing casts, hence using the `tryCast()` function to abstract the operation, which seems to be calling `evaluateCast()`.

Reply:
Good point. I suppose we need to refactor testCast() to use evaluateOnce() internally first. Or we could provide a testCast specialization that takes a single values.

Reply:
Right. `evaluateCast()` seems a bit more specialized, relying on the `CastTypedExpr` class to construct the cast expressions, so I'm not sure how practical it would be to refactor it to work with `evaluateOnce()`. 

I can add a convenience specialization to take in single values though, if you think we should go that way.